src/sentry/identity/vsts/provider.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+
+from django.http import HttpResponse
+
+from sentry import options
+from sentry.options.manager import FLAG_PRIORITIZE_DISK
+from sentry.identity.oauth2 import OAuth2Provider, OAuth2LoginView, OAuth2CallbackView
+from sentry.utils.pipeline import PipelineView
+
+options.register('vsts.client-id', flags=FLAG_PRIORITIZE_DISK)
+options.register('vsts.client-secret', flags=FLAG_PRIORITIZE_DISK)
+options.register('vsts.verification-token', flags=FLAG_PRIORITIZE_DISK)
+
+
+class VSTSIdentityProvider(OAuth2Provider):
+    key = 'vsts'
+    name = 'VSTS'
+
+    oauth_access_token_url = 'https://app.vssps.visualstudio.com/oauth2/token'
+    oauth_authorize_url = 'https://app.vssps.visualstudio.com/oauth2/authorize'
+
+    oauth_scopes = (
+        'vso.code_full',
+        'vso.identity_manage',
+        'vso.work_full',
+    )
+
+    def get_oauth_client_id(self):
+        return options.get('vsts.client-id')
+
+    def get_oauth_client_secret(self):
+        return options.get('vsts.client-secret')
+
+    def get_pipeline_views(self):
+        return [
+            OAuth2LoginView(
+                authorize_url=self.oauth_authorize_url,
+                client_id=self.get_oauth_client_id(),
+                scope=' '.join(self.get_oauth_scopes()),
+            ),
+            VSTSOAuth2CallbackView(
+                access_token_url=self.oauth_access_token_url,
+                client_id=self.get_oauth_client_id(),
+                client_secret=self.get_oauth_client_secret(),
+            ),
+            ConfigView(),
```

Comment:
Somewhat. In order to make any other requests than oauth, I need the account name (or server instance for on-prem). Unfortunately because the oauth doesn't give me anything else, I am having to ask the user for their account name (at the very least).